reframe/frontend/statistics.py

```diff
@@ -1,89 +1,103 @@
 import reframe.core.debug as debug
+from reframe.core.exceptions import (ReframeFatalError, StatisticsError)
 
 
 class TestStats:
     """Stores test case statistics."""
 
-    def __init__(self, tasks=[]):
-        if not hasattr(tasks, '__iter__'):
-            raise TypeError('expected an iterable')
-
-        # Store test cases per partition internally
-        self._tasks_bypart = {}
-        for t in tasks:
-            partition = t.check.current_partition
-            partname = partition.fullname if partition else 'None'
-            tclist = self._tasks_bypart.setdefault(partname, [])
-            tclist.append(t)
+    def __init__(self):
+        self._tasks = [[]]
```

Comment:
Please add a small comment as to what this nested list keeps.

Reply:
I would vote for `# Tasks per run stored as follows: [[run0_tasks], [run1_tasks], ...]`

Reply:
Done.